cmake/function/hypervisor_add_info.cmake

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (C) 2020 Assured Information Security, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+include(${bsl_SOURCE_DIR}/cmake/colors.cmake)
```

Comment:
There are numerous places like this where the hypervisor's build system is relying on the internal structure of the bsl project, and cmake files that belong to that project. I really think that this is going to end up causing unnecessary complexity, and will do more harm in the long run than the small amount of duplication you are trying to prevent.

Reply:
The hypervisor already is dependent on the internal structure of the BSL. All of the C++ code depends on this. For example: `#include <bsl/array.hypp>`. This is no different. All this means is that the location of CMake APIs is part of the BSL's provided APIs along with the C++ code. If this ever becomes an issue, we can address it then, for now I say we move on.